src/test/scala/RegexTokenizerTest.scala

```diff
@@ -0,0 +1,182 @@
+    " src=127.127.127.127 dst=127.0.0.1 src_port=52362" +
+    " dst_port=8151 session_id=0 reason=Traffic Denied"
+
+  val amount: Long = 10
+
+  val testSchema: StructType = new StructType(
+    Array[StructField]
+      (StructField("_time", TimestampType, nullable = false, new MetadataBuilder().build),
+        StructField("_raw", StringType, nullable = false, new MetadataBuilder().build),
+        StructField("index", StringType, nullable = false, new MetadataBuilder().build),
+        StructField("sourcetype", StringType, nullable = false, new MetadataBuilder().build),
+        StructField("host", StringType, nullable = false, new MetadataBuilder().build),
+        StructField("source", StringType, nullable = false, new MetadataBuilder().build),
+        StructField("partition", StringType, nullable = false, new MetadataBuilder().build),
+        StructField("offset", LongType, nullable = false, new MetadataBuilder().build),
+        StructField("estimate(tokens)", LongType, nullable = false, new MetadataBuilder().build)
+      )
+  )
+
+  @org.junit.jupiter.api.Test
+  def testTokenization(): Unit = {
+    val sparkSession = SparkSession.builder.master("local[*]").getOrCreate
+    val sqlContext = sparkSession.sqlContext
+    sparkSession.sparkContext.setLogLevel("ERROR")
+    val encoder = RowEncoder.apply(testSchema)
+    val rowMemoryStream = new MemoryStream[Row](1,sqlContext)(encoder)
+
+    var rowDataset = rowMemoryStream.toDF
+    val regex: String = "\\d+"
+    // create Scala udf for tokenizer
+    val regexUDF = functions.udf(new RegexTokenizerUDF, DataTypes.createArrayType(DataTypes.BinaryType, false))
+    // register tokenizer udf
+    sparkSession.udf.register("regex_udf", regexUDF)
+    // create pattern col for regex
+    rowDataset = rowDataset.withColumn("regex", functions.lit(regex))
+    // apply tokenizer udf to column
+    rowDataset = rowDataset.withColumn("tokens", regexUDF.apply(functions.col("_raw"), functions.col("regex")))
+
+    // create Scala udf for ByteArrayListasStringList
+    val byteArrayListAsStringListUDF = functions.udf(new ByteArrayListAsStringListUDF, DataTypes.createArrayType(StringType))
+    sparkSession.udf.register("bytes_to_string_udf", byteArrayListAsStringListUDF)
+    rowDataset = rowDataset.withColumn("tokensAsStrings", byteArrayListAsStringListUDF.apply(functions.col("tokens")))
+
+    val streamingQuery = startStream(rowDataset)
+    var run: Long = 0
+
+    while (streamingQuery.isActive) {
+      val time = Timestamp.valueOf(LocalDateTime.ofInstant(Instant.now, ZoneOffset.UTC))
+      rowMemoryStream.addData(
+        makeRows(time, String.valueOf(run)))
+
+      run += 1
+
+      if (run == 10) {
+        streamingQuery.processAllAvailable
+        streamingQuery.stop
+        streamingQuery.awaitTermination()
+      }
+    }
+
+    val resultCollected = sqlContext.sql("SELECT tokensAsStrings FROM TokenAggregatorQuery").collect()
+
```

Comment:
Assert that resultCollected has expected amount of results so the for loop executes as expected, otherwise theres a chance it will never be called and test passes

Reply:
added assert